.ci/Jenkinsfile

```diff
@@ -334,13 +325,43 @@ def checkSkipTests() {
 }
 
 
-def sshexec(workspace, connection, cmd){
-  sh "ssh -tt -o TCPKeepAlive=yes -o ServerAliveInterval=60 -o ServerAliveCountMax=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${workspace}/e2essh ${connection.user}@${connection.ip} -- '${cmd}'"
+def sshexec(userAtIP, cmd){
+  sh "ssh -tt -o TCPKeepAlive=yes -o ServerAliveInterval=60 -o ServerAliveCountMax=10 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${E2E_SSH_KEY} ${userAtIP} -- '${cmd}'"
 }
 
-def scpr(workspace, connection, remote_src, local_dst){
+def scpr(userAtIP, remote_src, local_dst){
   retryWithSleep(retries: 3, seconds: 5, backoff: true){
-    sh "scp -r -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${workspace}/e2essh ${connection.user}@${connection.ip}:${remote_src} ${local_dst}"
+    sh "scp -r -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i ${E2E_SSH_KEY} ${userAtIP}:${remote_src} ${local_dst}"
+  }
+}
+
+/*
+ * Runs the Make build at the CI, executing the closure in the context of Ansible + AWS
+ */
+def ciBuild(Closure body){
+  withEnv([
+    "SSH_KEY=${E2E_SSH_KEY}"
+  ]) {
+    def awsProps = getVaultSecret(secret: "${AWS_PROVISIONER_SECRET}")
+    def awsAuthObj = awsProps?.data
+    withEnv([
+      "ANSIBLE_CONFIG=${env.WORKSPACE}/.ci/ansible/ansible.cfg",
+      "ANSIBLE_HOST_KEY_CHECKING=False",
+    ]){
+      withVaultToken(){
+        withEnvMask(vars: [
+          [var: "AWS_ACCESS_KEY_ID", password: awsAuthObj.access_key],
+          [var: "AWS_SECRET_ACCESS_KEY", password: awsAuthObj.secret_key]
+        ]) {
+          retryWithSleep(retries: 3, seconds: 5, backoff: true){
+            withOtelEnv() {
+              sh("make -C .ci setup-env") // make sure the environment is created
```

Comment:
The .venv will be transferred to the nodes in the `stash` built-in step. It increases the stash/unstash time, so we could improve that. 

OTOH, the nodes will share python env